### PR TITLE
Migrate old uses of manual autocheck to use the new prepend autocheck

### PR DESCRIPTION
**Pull request by cgranleese-r7**
These changes have been made in reference to https://github.com/rapid7/metasploit-framework/pull/13787.

I have migrated the old uses of manual check calls to use the new prepend autocheck approach.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] Test some of the changed modules e.g. `use exploit/android/local/futex_requeue`
- [ ] **Verify** that changed modules operate correctly

### CODE REVIEW

File: modules/exploits/linux/local/diamorphine_rootkit_signal_priv_esc.rb [hunk 1]
```diff
@@ -88,12 +88,6 @@ def check
   end
 
   def exploit
-    if is_root?
-      unless datastore['ForceExploit']
-        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.'
-      end
-    end
```
Comment (space-r7) [inline modules/exploits/linux/local/diamorphine_rootkit_signal_priv_esc.rb:95]: The `is_root?` check should remain here.

File: modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb [hunk 2]
```diff
@@ -221,12 +221,6 @@ def check
   end
 
   def exploit
-    if is_root?
-      unless datastore['ForceExploit']
-        fail_with(Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.')
-      end
-    end
```
Comment (space-r7) [inline modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb:228]: The `is_root?` check should remain.

**Review by space-r7 — COMMENTED** [review]
There are just a couple of modules that need to keep their checks for elevated privileges. Everything else looks good to me.

File: modules/exploits/linux/local/diamorphine_rootkit_signal_priv_esc.rb (hunk 1 above)
Comment (cgranleese-r7) [reply modules/exploits/linux/local/diamorphine_rootkit_signal_priv_esc.rb:95]: Ok, I will be sure to get that added back in.

File: modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb (hunk 2 above)
Comment (cgranleese-r7) [reply modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb:228]: Ok, I will be sure to get that added back in.

**Comment by bcoles** [comment]
> 
> 
> There are just a couple of modules that need to keep their checks for elevated privileges. Everything else looks good to me.

Agreed. I took a look over the changes, in particular the modules I was involved with in some way, and the changes look good.

To offer some context around `is_root?`, the idea was to prevent operators running modules unnecessarily, which would needlessly impact opsec and waste time. However, it is also entirely possible, although unlikely, that a user will want to run a module even when `is_root?` detects that the session has root privileges. In particular, when attempting to break out of user namespaces, break out of containers, break out of virtual machines, or in the event that `is_root?` fails horribly somehow.

**Comment by space-r7** [comment]
So I started testing and realized that one of the modules changed is an auxiliary module. Given that auxiliary modules use `run` as opposed to `exploit`, using the `AutoCheck` mixin with them won't give the expected results.`check` methods are pretty common in auxiliary modules nowadays, so maybe it would make sense to implement an auxiliary-module specific `AutoCheck` mixin or modify the existing `AutoCheck` mixin.

On the other hand, there is only one auxiliary module that uses a manual `AutoCheck` currently, and I'm uncertain as to whether or not devs will see a need in using this functionality in their auxiliary modules in the future. Any other opinions on this?

**Comment by cgranleese-r7** [comment]
Just to make everyone aware, I will be at training for the next two weeks. So I'll be back on 2nd October. Incase I don't get back to any comments/make any progress on suggested changes etc.

**Comment by adfoster-r7** [comment]
@space-r7 That's a great spot, thank you! I don't have strong opinions here, maybe @wvu-r7 has some thoughts here?

**Comment by wvu-r7** [comment]
https://github.com/rapid7/metasploit-framework/pull/14040#discussion_r477262839

**Comment by adfoster-r7** [comment]
Thanks @wvu !

Closing this PR for now to keep the PR queue number down, and we'll reopen this after the above comment has been resolved - in one form or another :+1:

**Comment by adfoster-r7** [comment]
Just an update on this:

The AutoCheck module added support for Auxiliary modules: https://github.com/rapid7/metasploit-framework/pull/14458. This effort will be revived again in the new year 📈

File: modules/exploits/windows/local/ricoh_driver_privesc.rb [hunk 3]
```diff
@@ -140,11 +138,6 @@ def exploit
       fail_with(Failure::BadConfig, 'The payload should use the same architecture as the target driver')
     end
 
-    @driver_path = ''
```
Comment (adfoster-r7) [inline modules/exploits/windows/local/ricoh_driver_privesc.rb:143]: I don't think this line should be deleted 👀

Comment (cgranleese-r7) [reply modules/exploits/windows/local/ricoh_driver_privesc.rb:143]: Got that added back in 👍 , sorry about that.

Comment (adfoster-r7) [reply modules/exploits/windows/local/ricoh_driver_privesc.rb:143]: Oh closer inspection, it looks like the check method mutates this value - so deleting this assignment here works :+1:

I'm sure I knew this at the time 😄

Comment (cgranleese-r7) [reply modules/exploits/windows/local/ricoh_driver_privesc.rb:143]: Got that taken back out 😄

**Comment by space-r7** [comment]
Removed a small block from `emacs_movemail` in commit 606c6561a00e23fa6e7bb72b1519f93d6d6cd2d0. Everything else looked okay! Thanks, @cgranleese-r7!

**Comment by space-r7** [comment]
## Release Notes

This ensures that all modules that previously used manual `AutoCheck` behavior now leverage the `AutoCheck` mixin instead.

**Comment by wvu-r7** [comment]
Great work here. Thanks for taking care of this!

**Pull request merged**